### PR TITLE
vk: Refactor render target code

### PR DESCRIPTION
**Pull request by kd-11**
Base commit for a longer sequence of completed works

### CODE REVIEW

File: rpcs3/Emu/RSX/VK/VKRenderTargets.cpp [hunk 1]
```diff
@@ -0,0 +1,565 @@
...
+			cmd, resolve_surface->value,
+			resolve_surface->current_layout, VK_IMAGE_LAYOUT_GENERAL,
+			VK_PIPELINE_STAGE_TRANSFER_BIT | VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT,
+			VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT,
+			VK_ACCESS_TRANSFER_READ_BIT | VK_ACCESS_SHADER_READ_BIT,
+			VK_ACCESS_SHADER_WRITE_BIT,
+			range);
+
+		this->current_layout = VK_IMAGE_LAYOUT_GENERAL;
+		resolve_surface->current_layout = VK_IMAGE_LAYOUT_GENERAL;
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKRenderTargets.cpp:71]: indentation

File: rpcs3/Emu/RSX/VK/VKRenderTargets.cpp [hunk 2]
```diff
@@ -0,0 +1,565 @@
...
+			cmd, resolve_surface->value,
+			resolve_surface->current_layout, VK_IMAGE_LAYOUT_GENERAL,
+			VK_PIPELINE_STAGE_TRANSFER_BIT,
+			VK_PIPELINE_STAGE_COMPUTE_SHADER_BIT,
+			VK_ACCESS_TRANSFER_WRITE_BIT,
+			VK_ACCESS_SHADER_READ_BIT,
+			range);
+
+		this->current_layout = VK_IMAGE_LAYOUT_GENERAL;
+		resolve_surface->current_layout = VK_IMAGE_LAYOUT_GENERAL;
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKRenderTargets.cpp:144]: indentation

File: rpcs3/Emu/RSX/VK/VKRenderTargets.cpp [hunk 3]
```diff
@@ -0,0 +1,565 @@
...
+		{
+			// Underlying memory has been modified and we could not find valid data to fill it
+			clear_rw_barrier();
+
+			state_flags |= rsx::surface_state_flags::erase_bkgnd;
+			initialize_memory(cmd, read_access);
+			ensure(state_flags == rsx::surface_state_flags::ready);
+		}
+
+			// NOTE: Optimize flag relates to stencil resolve/unresolve for NVIDIA.
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKRenderTargets.cpp:556]: indentation

File: rpcs3/Emu/RSX/VK/VKRenderTargets.h [hunk 4]
```diff
@@ -21,582 +21,38 @@ namespace vk
...
-			// NOTE: Optimize flag relates to stencil resolve/unresolve for NVIDIA.
-			on_write_copy(newest_tag, optimize_copy);
-
-			if (!read_access && samples() > 1)
-			{
-				// Write barrier, must initialize
-				unresolve(cmd);
-			}
-		}
+			VkImageAspectFlags mask = VK_IMAGE_ASPECT_COLOR_BIT | VK_IMAGE_ASPECT_DEPTH_BIT) override;
```
Comment (Megamouse) [inline rpcs3/Emu/RSX/VK/VKRenderTargets.h:51]: indentation

**Review by Megamouse — COMMENTED** [review]
I wonder how much of the code is copy pasta (don't wanna review copy pasta (at least as far as I can review the GPU code))

**Comment by kd-11** [comment]
> I wonder how much of the code is copy pasta (don't wanna review copy pasta (at least as far as I can review the GPU code))

All of it lol. It's just moving code from .h to .cpp

**Comment by Megamouse** [comment]
then it should be fine after all the auto-indentation errors were resolved.
Maybe it is worth looking at includes in the header now.
If we're lucky we can move something to the cpp

File: rpcs3/VKGSRender.vcxproj.filters [hunk 5]
```diff
@@ -61,10 +61,11 @@
     </ClCompile>
     <ClCompile Include="Emu\RSX\VK\vkutils\image_helpers.cpp">
       <Filter>vkutils</Filter>
-    </ClCompile>
+    </ClCompile>
     <ClCompile Include="Emu\RSX\VK\VKOverlays.cpp" />
-    <ClCompile Include="Emu\RSX\VK\VKCompute.cpp" />
-    <ClCompile Include="Emu\RSX\VK\VKAsyncScheduler.cpp" />
+    <ClCompile Include="Emu\RSX\VK\VKCompute.cpp" />
+    <ClCompile Include="Emu\RSX\VK\VKAsyncScheduler.cpp" />
```
Comment (Megamouse) [inline rpcs3/VKGSRender.vcxproj.filters:67]: suspicious

File: rpcs3/Emu/RSX/VK/VKRenderTargets.h (hunk 4 above)
Comment (Megamouse) [reply rpcs3/Emu/RSX/VK/VKRenderTargets.h:51]: ups. this one seems to be fine. sorry

File: rpcs3/VKGSRender.vcxproj.filters (hunk 5 above)
Comment (kd-11) [reply rpcs3/VKGSRender.vcxproj.filters:67]: visual studio works in mysterious ways

Comment (Megamouse) [reply rpcs3/VKGSRender.vcxproj.filters:67]: The last person before must have done a manual edit.

**Comment by kd-11** [comment]
That's still low prio for me, there's still a fairly severe leak happening somewhere between GPU and CPU, and I don't have it 100% patched. I've been picking at it for something like 100h by now and I keep finding bugs and patching them, but new problems just show up with every fix. Anyway, I'm sitting on quite a lot of code related to this stuff, I just want to start committing stuff slowly while I keep tracking down the leak. "Lucky" for me, the test scenario has random addresses each time, making debugging the problem ~~super fun~~ incredibly stressful.

**Pull request merged**